sessions-core/src/main/java/com/adyen/checkout/sessions/repository/SessionRepository.kt

```diff
@@ -0,0 +1,184 @@
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by josephj on 17/3/2022.
+ */
+
+package com.adyen.checkout.sessions.repository
+
+import com.adyen.checkout.components.ActionComponentData
+import com.adyen.checkout.components.api.suspendedCall
+import com.adyen.checkout.components.base.Configuration
+import com.adyen.checkout.components.model.payments.request.OrderRequest
+import com.adyen.checkout.components.model.payments.request.PaymentComponentData
+import com.adyen.checkout.components.model.payments.request.PaymentMethodDetails
+import com.adyen.checkout.core.exception.CheckoutException
+import com.adyen.checkout.core.log.LogUtil
+import com.adyen.checkout.core.log.Logger
+import com.adyen.checkout.sessions.api.SessionBalanceConnection
+import com.adyen.checkout.sessions.api.SessionCancelOrderConnection
+import com.adyen.checkout.sessions.api.SessionCreateOrderConnection
+import com.adyen.checkout.sessions.api.SessionDetailsConnection
+import com.adyen.checkout.sessions.api.SessionPaymentsConnection
+import com.adyen.checkout.sessions.api.SessionSetupConnection
+import com.adyen.checkout.sessions.model.Session
+import com.adyen.checkout.sessions.model.request.SessionBalanceRequest
+import com.adyen.checkout.sessions.model.request.SessionCancelOrderRequest
+import com.adyen.checkout.sessions.model.request.SessionDetailsRequest
+import com.adyen.checkout.sessions.model.request.SessionOrderRequest
+import com.adyen.checkout.sessions.model.request.SessionPaymentsRequest
+import com.adyen.checkout.sessions.model.request.SessionSetupRequest
+import com.adyen.checkout.sessions.model.response.SessionBalanceResponse
+import com.adyen.checkout.sessions.model.response.SessionCancelOrderResponse
+import com.adyen.checkout.sessions.model.response.SessionDetailsResponse
+import com.adyen.checkout.sessions.model.response.SessionOrderResponse
+import com.adyen.checkout.sessions.model.response.SessionPaymentsResponse
+import com.adyen.checkout.sessions.model.response.SessionSetupResponse
+import java.io.IOException
+import org.json.JSONException
+
+class SessionRepository {
+
+    companion object {
+        private val TAG = LogUtil.getTag()
+    }
+
+    suspend fun setupSession(
+        configuration: Configuration,
+        session: Session,
+        order: OrderRequest?
+    ): SessionSetupResponse {
+        Logger.d(TAG, "Setting up session")
+        try {
+            val request = SessionSetupRequest(session.sessionData.orEmpty(), order)
+            return SessionSetupConnection(
+                request = request,
+                environment = configuration.environment,
+                sessionId = session.id,
+                clientKey = configuration.clientKey
+            ).suspendedCall()
+        } catch (e: IOException) {
+            Logger.e(TAG, "SessionSetupConnection Failed", e)
+            throw CheckoutException("Unable to setup session")
```

Comment:
So I've been thinking we should start having some better error handling in general, this looks like a good place to start, can we have a chat about it?

Reply:
Yes indeed, let's discuss it and make another PR to refactor all connections